Tests/DataPipeline/DataPipelineEventBustTests.swift

```diff
@@ -0,0 +1,81 @@
+@testable import CioDataPipelines
+@testable import CioInternalCommon
+import Foundation
+@testable import Segment
+@testable import SharedTests
+import XCTest
+
+class DataPipelineEventBustTests: IntegrationTest {
+    var outputReader: OutputReaderPlugin!
+
+    private var eventBusHandler: EventBusHandler {
+        diGraphShared.eventBusHandler
+    }
+
+    private let deviceAttributesMock = DeviceAttributesProviderMock()
+    private let globalDataStoreMock = GlobalDataStoreMock()
+    private var userAgentUtil: UserAgentUtil?
+
+    override func setUpDependencies() {
+        super.setUpDependencies()
+
+        diGraphShared.override(value: deviceAttributesMock, forType: DeviceAttributesProvider.self)
+        diGraphShared.override(value: globalDataStoreMock, forType: GlobalDataStore.self)
+    }
+
+    override func setUp() {
+        super.setUp(modifySdkConfig: nil)
+        // OutputReaderPlugin helps validating interactions with analytics
+        outputReader = (customerIO.add(plugin: OutputReaderPlugin()) as? OutputReaderPlugin)
+        userAgentUtil = UserAgentUtilImpl(deviceInfo: deviceInfoStub)
+    }
+
+    func testSubscribeToJourneyEvents_DataPipelineHandlesTrackMetricEvent() async {
+        diGraphShared.override(value: eventBusHandler, forType: EventBusHandler.self)
```

Comment:
Why we didn't do this in `setUpDependencies`?

Reply:
good catch, it was added by mistake